### ISSUE TITLE
Replace Filesystem, FTP with FlySystem

### ISSUE THREAD

**Issue by oparoz** [Codeception/Codeception#2271]
[FlySystem](http://flysystem.thephpleague.com/) is a filesystem abstraction which allows you to easily swap out a local filesystem for a remote one. Reducing technical debt and chance of vendor lock-in.

It could replace the custom FileSystem and FTP modules and offer support for many more filesystems. Here is a list of what's available today :
* Local
* Azure
* AWS S3 V2
* AWS S3 V3
* Copy.com
* Dropbox
* FTP / SFTP
* GridFS
* Memory
* Rackspace
* WebDAV
* PHPCR
* ZipArchive

**Comment by DavertMik** [comment]
Good idea. If you can get this module implemented that would be really cool!

**Comment by pavelsavelyev** [comment]
Hi, guys! We developed our version of the module and share it https://github.com/lamoda/codeception-flysystem currently supports 2 protocols (FTP, WebDAV), we already using it in our tests and will be happy if it will be useful for you. Comments/issues are welcome too :)

**Comment by SamMousa** [comment]
@pavelsavelyev it looks like something we could take inspiration from!

It doesn't have any tests though...

**Comment by pavelsavelyev** [comment]
@SamMousa Thank you for your feedback :) We will add tests in the nearest future, already created an issue based on your advice  https://github.com/lamoda/codeception-flysystem/issues/1

**Comment by DavertMik** [comment]
Thanks, @pavelsavelyev 
Wow, just watched video from Badoo Meetup where this module was presented by your colleague.
Looks awesome, I added it to our addons page https://codeception.com/addons

Probably we could have some chapter on filesystem testing in our documentation but I can't figure out where to put it. Maybe in Data testing? Maybe not )

**Comment by pavelsavelyev** [comment]
Oh, great news, really happy to see the module on Add-ons page, thank you @DavertMik   :) 
In the nearest future, we extend supported formats by S3 (this sprint). And after that probably we will share a couple more modules. 
About where to put documentation, currently best options seems page Data, but not sure - everything is data :) and this page may become too big in perspective.
